pycalphad/mapping/compat_api.py

```diff
@@ -0,0 +1,111 @@
+    return_strategy : bool, optional
+        Return the BinaryStrategy object in addition to the Axes. Defaults to False.
+    map_kwargs : dict, optional
+        Additional keyword arguments to BinaryStrategy().
+    plot_kwargs : dict, optional
+        Keyword arguments to plot_binary().
+
+    Returns
+    -------
+    Axes
+        Matplotlib Axes of the phase diagram
+    (Axes, BinaryStrategy)
+        If return_strategy is True.
+
+    """
+    indep_comps = [key for key, value in conditions.items() if not map_utils.is_state_variable(key) and len(np.atleast_1d(value)) > 1]
+    indep_pots = [key for key, value in conditions.items() if map_utils.is_state_variable(key) and len(np.atleast_1d(value)) > 1]
+    if (len(indep_comps) != 1) or (len(indep_pots) != 1):
+        raise ValueError('binplot() requires exactly one composition coordinate and one potential coordinate')
+
+    strategy = BinaryStrategy(database, components, phases, conditions, **map_kwargs)
+    strategy.initialize()
+    strategy.do_map()
+
+    plot_kwargs = plot_kwargs if plot_kwargs is not None else dict()
+    ax = plot_binary(strategy, **plot_kwargs)
+    ax.grid(plot_kwargs.get("gridlines", False))
+
+    if return_strategy:
+        return ax, strategy
+    else:
+        return ax
+
+
+def ternplot(dbf, comps, phases, conds, x=None, y=None, return_strategy=False, map_kwargs=None, **plot_kwargs):
+    """
+    Calculate the ternary isothermal, isobaric phase diagram.
+
+    Parameters
+    ----------
+    dbf : Database
+        Thermodynamic database containing the relevant parameters.
+    comps : Sequence[str]
+        Names of components to consider in the calculation.
+    phases : Sequence[str]
+        Names of phases to consider in the calculation.
+    conds : Mapping[v.StateVariable, Union[float, Tuple[float, float, float]]]
+        Maps StateVariables to values and/or iterables of values.
+        For ternplot only two changing composition coordinates is supported.
+    x : v.MoleFraction
+        instance of a pycalphad.variables.composition to plot on the x-axis.
+        Must correspond to an independent condition.
+    y : v.MoleFraction
+        instance of a pycalphad.variables.composition to plot on the y-axis.
+        Must correspond to an independent condition.
+    return_strategy : bool, optional
+        Return the TernaryStrategy object in addition to the Axes. Defaults to False.
+    map_kwargs : dict, optional
+        Additional keyword arguments to TernaryStrategy().
+    plot_kwargs : dict, optional
+        Keyword arguments to plot_ternary().
+
+    Returns
+    -------
+    Axes
+        Matplotlib Axes of the phase diagram
+    (Axes, TernaryStrategy)
+        If return_strategy is True.
+
+    """
+    # remaining plot_kwargs from pycalphad.plot.eqplot
+    # x=None, y=None, z=None, tieline_color=(0, 1, 0, 1), tie_triangle_color=(1, 0, 0, 1), **kwargs
+    # kwargs passed ot ax.scatter
```

Comment:
I didn't even notice the z variable this whole time. 

One thing that always bugged me between binplot and ternplot was binplot used (plot_kwargs = None, **map_kwargs) and ternplot used (map_kwargs=None, **plot_kwargs) in order to retain the same API in the previous binplot and ternplot functions. It seemed inconsistent to have to input a dictionary to change the tieline colors in binplot while I only needed an keyword argument for ternplot. I suppose it's okay though if we're moving away from binplot/ternplot and towards a `Renderer` API.

Regarding this difference, ternplot will have `tieline_color` and `tie_triangle color` as arguments to the docstring, but binplot will mention these in the docstrings as variables to be added to the plot_kwargs dictionary. I will note that tie_triangle_color is a new argument for binplot since we can plot invariants now.

Reply:
I'm not sure how I managed to make those perfectly inconsistent, but I'm happy with keeping it how it is for now, to maintain compatibility with existing code